packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts

```diff
@@ -201,34 +213,48 @@ export class FirelensLogRouter extends ContainerDefinition {
     super(scope, id, props);
     const options = props.firelensConfig.options;
     if (options) {
+      const hasConfig = (options.configFileValue !== undefined);
       const enableECSLogMetadata = options.enableECSLogMetadata || options.enableECSLogMetadata === undefined;
       const configFileType = (options.configFileType === undefined || options.configFileType === FirelensConfigFileType.S3) &&
-        (cdk.Token.isUnresolved(options.configFileValue) || /arn:aws[a-zA-Z-]*:s3:::.+/.test(options.configFileValue))
+      (cdk.Token.isUnresolved(options.configFileValue) || /arn:aws[a-zA-Z-]*:s3:::.+/.test(options.configFileValue || ''))
```

Comment:
Because of compile time checks IIRC, the `test` call can't be made on undefined... I know that the bit before should be false if it is undefined but last time the build failed without this... I don't really understand typescript...

Reply:
Actually now that I installed typescript plugin VS code actually flags it...

Reply:
It does indeed fail the build... it feels like the typescript compiler isn't actually that smart in figuring out if a value will be undefined or not

```
@aws-cdk/aws-ecs: lib/firelens-log-router.ts:226:92 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
@aws-cdk/aws-ecs:   Type 'undefined' is not assignable to type 'string'.
@aws-cdk/aws-ecs: 226       (cdk.Token.isUnresolved(options.configFileValue) || /arn:aws[a-zA-Z-]*:s3:::.+/.test(options.configFileValue))
```